src/NuGet.Core/NuGet.Test.Utility/CommandRunner.cs

```diff
@@ -58,36 +59,19 @@ public class CommandRunner
             {
                 p = new Process();
 
-                p.OutputDataReceived += (o, e) =>
-                {
-                    if (e.Data != null)
-                    {
-                        output.AppendLine(e.Data);
-                    }
-                };
-
-                p.ErrorDataReceived += (o, e) =>
-                {
-                    if (e.Data != null)
-                    {
-                        errors.AppendLine(e.Data);
-                    }
-                };
-
                 p.StartInfo = psi;
                 p.Start();
 
-                p.BeginErrorReadLine();
-                p.BeginOutputReadLine();
+                var outputTask = ConsumeStreamReaderAsync(p.StandardOutput, output);
+                var errorTask = ConsumeStreamReaderAsync(p.StandardError, errors);
```

Comment:
Yes, needs to be on another thread. See the remarks of `Process.StandardOutput`:
https://msdn.microsoft.com/en-us/library/system.diagnostics.process.standardoutput(v=vs.110).aspx#Anchor_2

Note that the `WaitAll` is after the `WaitForExit`, so process is ending before output is required to complete. That being said, the if the process does not complete (due to timeout), we should not wait for output completion. I'll address this.